src/RA_Achievement.cpp

```diff
@@ -20,7 +20,8 @@ void _ReadStringTil(std::string& value, char nChar, const char*& pSource)
 
 //////////////////////////////////////////////////////////////////////////
 
-Achievement::Achievement(AchievementSetType nType) :
+_Use_decl_annotations_
+Achievement::Achievement(Type nType) noexcept :
```

Comment:
With the removal of `IsCoreAchievement`, `m_nSetType` should no longer be used, eliminating the need to pass `nType` in the constructor, which should allow you to put the enum directly in `AchievementSet` and clean up some of the stuff you've been trying to work around.